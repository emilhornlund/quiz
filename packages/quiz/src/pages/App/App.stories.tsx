import type { Meta, StoryObj } from '@storybook/react'

import App from './App.tsx'

const meta = {
  component: App,
} satisfies Meta<typeof App>

export default meta
type Story = StoryObj<typeof meta>

export const Example = {
  args: {
    primary: true,
    label: 'App',
  },
} satisfies Story
