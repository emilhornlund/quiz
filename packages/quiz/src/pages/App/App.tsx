import { MessageDto } from '@quiz/common'
import React, { FC, useState } from 'react'

import viteLogo from '/vite.svg'

import reactLogo from '../../assets/react.svg'

import styles from './App.module.scss'

const App: FC = () => {
  const [message, setMessage] = useState<string>()

  const handleClickFetchMessage = () => {
    fetch('/quiz-service/api/hello')
      .then((response) => {
        if (response.ok) {
          return response.json()
        }
        throw new Error('Api Error')
      })
      .then((data) => data as MessageDto)
      .then(({ value }) => value)
      .then(setMessage)
  }

  return (
    <div className={styles.main}>
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src={viteLogo} className={styles.logo} alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank" rel="noreferrer">
          <img
            src={reactLogo}
            className={[styles.logo, styles.react].join(' ')}
            alt="React logo"
          />
        </a>
      </div>
      <h1>Vite + React</h1>
      <div className={styles.card}>
        <button onClick={handleClickFetchMessage} data-testid="fetch-button">
          {message ?? 'Fetch'}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className={styles.readTheDocs}>
        Click on the Vite and React logos to learn more
      </p>
    </div>
  )
}

export default App
