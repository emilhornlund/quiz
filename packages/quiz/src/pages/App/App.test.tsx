import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import React from 'react'
import { expect, test } from 'vitest'

import App from './App.tsx'

test('should render application', async () => {
  const { container } = render(<App />)

  expect(container).toMatchSnapshot()
})

test('should fetch hello world message', async () => {
  const { container } = render(<App />)

  screen.getByText('Fetch')

  const button = screen.getByTestId('fetch-button')
  fireEvent.click(button)

  await waitFor(() => {
    expect(screen.getByText('Hello, World!')).toBeInTheDocument()
  })

  expect(container).toMatchSnapshot()
})
