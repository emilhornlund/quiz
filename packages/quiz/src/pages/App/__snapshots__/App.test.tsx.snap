// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should fetch hello world message 1`] = `
<div>
  <div
    class="main"
  >
    <div>
      <a
        href="https://vitejs.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/vite.svg"
        />
      </a>
      <a
        href="https://react.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react"
          src="/src/assets/react.svg"
        />
      </a>
    </div>
    <h1>
      Vite + React
    </h1>
    <div
      class="card"
    >
      <button
        data-testid="fetch-button"
      >
        Hello, World!
      </button>
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to test HMR
      </p>
    </div>
    <p
      class="readTheDocs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</div>
`;

exports[`should render application 1`] = `
<div>
  <div
    class="main"
  >
    <div>
      <a
        href="https://vitejs.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/vite.svg"
        />
      </a>
      <a
        href="https://react.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react"
          src="/src/assets/react.svg"
        />
      </a>
    </div>
    <h1>
      Vite + React
    </h1>
    <div
      class="card"
    >
      <button
        data-testid="fetch-button"
      >
        Fetch
      </button>
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to test HMR
      </p>
    </div>
    <p
      class="readTheDocs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>
</div>
`;
